src/index.ts

```diff
@@ -1,112 +1,84 @@
+	);
+});
 
 router.get`/callback`.handle(async (ctx) => {
-  let url = new URL(ctx.event.request.url)
-  let code = url.searchParams.get('code')
-
-  if (!code) {
-    return ctx.end('Missing authorization code', { status: 400 })
-  }
-
-  let accessTokenResp = await getAccessToken(code)
-
-  if (accessTokenResp.status !== 200) {
-    return ctx.end(
-      `${
-        accessTokenResp.status
-      } error from Discord: ${await accessTokenResp.text()}`,
-      { status: 400 },
-    )
-  }
-
-  let accessToken = (await accessTokenResp.json())['access_token']
-
-  let userGuilds = await getUserGuilds(accessToken)
-
-  let isAllowed = false
-
-  for (var guild of WHITELISTED_GUILDS) {
-    if (userGuilds.indexOf(guild) !== -1) {
-      isAllowed = true
-    }
-  }
-
-  if (!isAllowed) {
-    return ctx.end('You are not authorized to join DDD.', { status: 403 })
-  }
-
-  let userDetails = await getUserDetails(accessToken)
-
-  let joinCode = await joinUserToGuild(accessToken, userDetails['id'])
-
-  let revokeResponse = await revokeToken(accessToken)
-
-  let msg = ''
-
-  if (revokeResponse.status !== 200) {
-    msg =
-      ' Your authorization token could not be revoked, you can do so in your Discord client under User Settings > Authorized Apps'
-  }
-
-  if (joinCode === 201 || joinCode === 204) {
-    return ctx.end(`Redirecting...`, {
-      status: 302,
-      headers: {
-        Location: LANDING_URL,
-      },
-    })
-  } else {
-    return ctx.end(
-      `An unexpected error occurred while joining you to the guild.${msg}`,
-    )
-  }
-})
-
-router.all`(.*)`.handle((ctx) => ctx.end('Not found', { status: 404 }))
-
-addEventListener('fetch', e => {
-  const res = router.getResponseForEvent(e).catch(
-    error =>
-      new Response(getErrorPageHTML(e.request, error), {
-        status: 500,
-        headers: {
-          'Content-Type': 'text/html',
-        },
-      }),
-  )
-
-  e.respondWith(res as any)
-})
+	const url = new URL(ctx.event.request.url);
+	const code = url.searchParams.get('code');
+
+	if (!code) {
+		return ctx.end('Missing authorization code', { status: 400 });
+	}
+
+	try {
+		const accessTokenReq = await getAccessToken(code);
+		const accessTokenBody = await accessTokenReq[0];
+		const accessToken = accessTokenBody['access_token'];
+		const userGuilds = await getUserGuilds(accessToken);
+
+		if (!WHITELISTED_GUILDS.some((x) => userGuilds.includes(x)))
+			return ctx.end('You are not authorized to join DDD.', { status: 403 });
+
+		const userDetails = await getUserDetails(accessToken);
+		const joinUserToGuildStatus = (await (await joinUserToGuild(accessToken, userDetails.id))[1]).status;
+		const revokeUserTokenResponse = await (await revokeToken(accessToken))[1];
+
+		if (!revokeUserTokenResponse.ok)
+			return ctx.end(
+				`An unexpected error occurred while joining you to the guild. Your authorization token could not be revoked, you can do so in your Discord client under User Settings > Authorized Apps`,
```

Comment:
Inaccurate per say, this error is if we couldn't revoke the token, not if we couldn't add you to a guild. Should handle that case too